src/CoreWCF.Primitives/src/CoreWCF/Configuration/ServiceModelServiceCollectionExtensions.cs

```diff
@@ -17,6 +19,25 @@ public static IServiceCollection AddServiceModelServices(this IServiceCollection
             {
                 throw new ArgumentNullException(nameof(services));
             }
+            services.AddSingleton<WrappingIServer>();
+            for (int i=0; i<services.Count; i++)
+            {
+                if (services[i].ServiceType == typeof(IServer))
+                {
+                    Type implType = services[i].ImplementationType;
+                    if (!services.Any(d => d.ServiceType == implType))
+                    {
+                        services.AddSingleton(implType);
+                    }
+                    services[i] = ServiceDescriptor.Singleton<IServer>((provider) =>
+                        {
+                            var originalIServer = (IServer)provider.GetRequiredService(implType);
+                            var wrappingServer = provider.GetRequiredService<WrappingIServer>();
+                            wrappingServer.InnerServer = originalIServer;
+                            return wrappingServer;
+                        });
+                }
+            }
```

Comment:
This is probably necessary to run something that causes the legacy WebHost.StartAsync() to throw after the server binds to an endpoint. But if you can create an extension method on IHostBuilder (which configures the generic host which exists in 2.1 and is the only host getting new features) that runs after ConfigureWebHost(), you could add an IHostedService that starts after the server and causes Host.StartAsync() to throw.

Reply:
I opened an issue so this doesn't get missed. The second release will target .NET 5+ (probably want to settle on a LTS release, but aiming to work on 5.0+) and I think that's the time to do this as I plan to make some api changes anyway.